util/legacy/proto_compatibility_test.go

```diff
@@ -0,0 +1,331 @@
+package legacy
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ovrclk/akash/app"
+	"github.com/ovrclk/akash/types"
+	dtypesv1beta1 "github.com/ovrclk/akash/x/deployment/legacy/types/v1beta1"
+	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	etypesv1beta1 "github.com/ovrclk/akash/x/escrow/legacy/types/v1beta1"
+	etypes "github.com/ovrclk/akash/x/escrow/types"
+	mtypesv1beta1 "github.com/ovrclk/akash/x/market/legacy/types/v1beta1"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	cdc = app.MakeEncodingConfig().Marshaler.(codec.BinaryCodec)
+)
+
+func TestDeployment_DeploymentProto_IsCompatible(t *testing.T) {
```

Comment:
This test indicates that v1beta1 `Deployment` proto serialization is compatible with v1beta2 `Deployment` proto.
It is because nothing changed inside the `Deployment` proto itself, instead only a import changed in the proto file, which is being used in some other type, not in the `Deployment` msg.

So, this indicates that we don't need the upgrade for `Deployment` proto.
I have included this upgrade ATM, but I think it should be removed as it is unnecessary.